src/CreeDictionary/CreeDictionary/management/commands/dumpimportjson.py

```diff
@@ -0,0 +1,332 @@
+                    slugs_to_grab.add(e["formOf"])
+                else:
+                    grabbed_slugs.add(e["slug"])
+        for e in self._entries:
+            if (
+                "slug" in e
+                and e["slug"] in slugs_to_grab
+                and e["slug"] not in grabbed_slugs
+            ):
+                test_entries.append(e)
+
+        if remaining_test_words:
+            raise Exception(
+                "Some entries from test_db_words.text were not found in the database: "
+                + ", ".join(w for w in sorted(remaining_test_words))
+            )
+
+        return json.dumps(test_entries, ensure_ascii=False, indent=2) + "\n"
+
+    def add_lemmas(self):
+        for wf in Wordform.objects.filter(is_lemma=True).prefetch_related(
+            "definitions__citations"
+        ):
+            paradigm = wf.paradigm or convert_crkeng_word_class_to_paradigm_name(
+                wf.word_class
+            )
+            if paradigm == "IPC":
+                paradigm = None
+
+            analysis = unsmush_analysis(wf.analysis)
+            # In importjson, headwords are only indexed for source-language
+            # search if analysis is None, on the presumption that anything with
+            # an analysis can be recognized by the relaxed analyzer FST.
+            #
+            # The XML import process makes up analyses for ‘as-is’ entries,
+            # making unanalyzable entries unsearchable. Leave these analyses
+            # out.
+            if wf.as_is:
+                analysis = None
+
+            entry = {
+                "head": wf.text,
+                "analysis": analysis,
+                "paradigm": paradigm,
+                "senses": [
+                    {
+                        "definition": d.text,
+                        "sources": [c.abbrv for c in d.citations.all()],
+                    }
+                    for d in wf.definitions.filter(
+                        auto_translation_source_id__isnull=True
+                    )
+                ],
+                "linguistInfo": {
+                    "as_is": wf.as_is,
+                    "inflectional_category": wf.inflectional_category,
+                    "inflectional_category_plain_english": LABELS.english.get(
+                        wf.inflectional_category
+                    ),
+                    "inflectional_category_linguistic": LABELS.linguistic_long.get(
+                        wf.inflectional_category
+                    ),
+                    "pos": wf.pos,
+                    "stem": wf.stem,
+                    "smushedAnalysis": wf.analysis,
+                    "wordclass": wf.wordclass_text,
+                    "wordclass_emoji": wf.get_emoji_for_cree_wordclass(),
+                },
+            }
+            if wf.inflectional_category == "IPV":
+                entry["definesFstTag"] = f"PV/{wf.text.rstrip('-')}+"
```

Comment:
I wrote some code for generating the LEXC content for all the preverbs in Arok's database, as that set is not entirely closed, though neither are additions going to be many. However, I haven't used that code yet, because ...

The one issue that I need to resolve is what to do with preverbs which are minimal pairs wrt vowel length, since in their case using the tag format `PV/xxx` ignoring vowel length results in ambiguity. Alternatives would be 1) using vowel length as is in the `PV/...` tags, which I've avoided due to a historical dislike for diacritics in tags; 2) using double vowel format to distinguish long-vowel preverbs in ambiguous cases, or 3) using a numeric index to distinguish ambiguous cases.

Reply:
In other words, the assumption made by this code is **not safe**. A `PV/*+` tag might be derived in some non-trivial way from its `IPV` entry. So this must be encoded in the import data instead ¯\\\_(ツ)\_/¯

Reply:
This field is not currently used by <https://github.com/UAlbertaALTLab/cree-intelligent-dictionary/pull/874>. But I do think it’s our best guess at what the contents of `definesFstTag` should be. We can update the import data when the morphodict code starts using it.